diff --git a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
index 5cd6c7bd344a..627b7a2fc619 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
@@ -176,6 +176,7 @@
 &audio {
 	pinctrl-names = "default";
 	pinctrl-0 = <&audio_pins>;
+	status = "okay";
 };
 
 &sdhost_gpio48 {
@@ -351,6 +352,22 @@
 	pinctrl-0 = <&i2s_pins>;
 };
 
+&fb {
+	status = "disabled";
+};
+ 
+&firmwarekms {
+	status = "okay";
+};
+
+&vc4 {
+	status = "okay";
+};
+
+&v3d {
+	status = "okay";
+};
+
 / {
 	__overrides__ {
 		act_led_gpio = <&act_led>,"gpios:4";
